frappe/public/js/frappe/list/list_view.js

```diff
@@ -1338,10 +1338,6 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 				return;
 			}
 
-			if (!frappe.get_doc(data?.doctype, data?.name)?.__unsaved) {
-				frappe.model.remove_from_locals(data.doctype, data.name);
```

Comment:
Hi @ankush,

for me the issue comes up in several occasions, the most stable way to reproduce it was:
1. Submit a Quotation
2. Go back to list view
3. Clear the cache
4. Open the submitted Quotation
5. Make it dirty (e.g. via More Info -> Print Settings -> Group same items)
6. Open the Details tab
7. Click on Update
See below video for reference.

https://user-images.githubusercontent.com/49870752/223674830-eea2f273-586e-496c-a905-0423ad0d46f1.mp4

> Keeping this does make sense because it immediately clears cache instead of waiting for user to open and perform check in background?
I am not super confident on everything that happens in the background but wouldn't the newer version of the doc be fetched anyway if the user opens the document? So the check if the document in locals is stale is done in any case, right?
> The fix instead could be reloading the doc if doc is not found.
I tried to reload the doc right after the above lines. It seems like it works fine.
So the alternative would be sth. like:
```
if (data.doctype === cur_doc.doctype && data.name === cur_doc.name) {
       cur_doc.reload_doc()
}
```
Or did you have sth. else in mind?

Best regards,
Leo

Reply:
This seems fine, as we mark the document stale anyway, and check for which gets triggered on document when it's loaded in form.